api/src/main/java/bio/terra/tanagra/service/search/Expression.java

```diff
@@ -35,13 +35,13 @@ public static Literal create(DataType dataType, String value) {
     }
   }
 
-  /** An {@link Expression} that's an {@link bio.terra.tanagra.service.search.Attribute} */
+  /** An {@link Expression} that's an {@link AttributeVariable}. */
   @AutoValue
   abstract class AttributeExpression implements Expression {
-    public abstract Attribute attribute();
+    public abstract AttributeVariable attributeVariable();
 
-    public static AttributeExpression create(Attribute attribute) {
-      return new AutoValue_Expression_AttributeExpression(attribute);
+    public static AttributeExpression create(AttributeVariable attributeVariable) {
```

Comment:
Is it necessary for all `AttributeExpressions` to be explicitly aliased to variables, or can expression be created with just an attribute and have the variable be implicitly generated (similar to the previous behaviour 'person.height` -> `person_89ab.height`?)

Reply:
It's currently necessary in the code/API as written. It's not necessary to keep that requirement, and if it's useful for clients we can add support for that. But it was easiest to start without implementing implicit variables.